src/auto_archiver/modules/gdrive_storage/gdrive_storage.py

```diff
@@ -93,27 +93,39 @@ def upload(self, media: Media, **kwargs) -> bool:
         # upload file to gd
         logger.debug(f"uploading {filename=} to folder id {upload_to}")
         file_metadata = {"name": [filename], "parents": [upload_to]}
-        media = MediaFileUpload(media.filename, resumable=True)
-        gd_file = (
-            self.service.files()
-            .create(supportsAllDrives=True, body=file_metadata, media_body=media, fields="id")
-            .execute()
-        )
-        logger.debug(f"uploadf: uploaded file {gd_file['id']} successfully in folder={upload_to}")
+
+        # DM 4th Jun 2025 - somehow a file was trying to be uploaded which didn't exist
+        # so catch and carry on instead of leaving in state: Archive in progress
+        try:
+            media = MediaFileUpload(media.filename, resumable=True)
+            gd_file = (
+                         self.service.files()
+                        .create(supportsAllDrives=True, body=file_metadata, media_body=media, fields="id")
+                        .execute()
+                    )
+            logger.debug(f"uploadf: uploaded file {gd_file['id']} successfully in folder={upload_to}")
+        except FileNotFoundError as e:
+            logger.error(f'gd uploadf: file not found {media.filename} - {e}')
+        except Exception as e:
+            logger.error(f'gd uploadf: error uploading {media.filename} to {upload_to} - {e}')
```

Comment:
I'm implementing this bug fix directly into /dev, no need for further PRs

Reply:
done https://github.com/bellingcat/auto-archiver/pull/312/commits/f53b2075ba61f21d7c80deba459a8e6c05c4cc80